core/common/src/test/java/alluxio/wire/TieredIdentityTest.java

```diff
@@ -64,6 +75,27 @@ public void thrift() {
   }
 
   @Test
+  public void locality() {
+    Configuration.set(PropertyKey.LOCALITY_COMPARE_NODE_IP, "true");
+
+    LocalityTier lt1 = new LocalityTier("node", "A");
+    LocalityTier lt2 = new LocalityTier("node", "A");
+    LocalityTier lt3 = new LocalityTier("node", "B");
+    LocalityTier lt4 = new LocalityTier("rack", "A");
+    LocalityTier lt5 = new LocalityTier("rack", "B");
+    LocalityTier lt6 = new LocalityTier("rack", "B");
+    LocalityTier lt7 = new LocalityTier("rack", "");
+    LocalityTier lt8 = new LocalityTier("node", "A");
+    LocalityTier lt9 = new LocalityTier("node", "");
+    assertTrue(lt1.matches(lt1));
+    assertTrue(lt1.matches(lt2));
+    assertFalse(lt2.matches(lt3));
+    assertTrue(lt5.matches(lt6));
+    assertFalse(lt4.matches(lt5));
+    assertFalse(lt6.matches(lt7));
+    assertFalse(lt8.matches(lt9));
+  }
+
   public void string() {
```

Comment:
We can add this test since most systems resolve localhost to 127.0.0.1. Systems where this isn't true can skip the test
```java
  @Test
  public void matchByIpResolution() throws Exception {
    assumeTrue(InetAddress.getByName("localhost").getHostAddress().equals("127.0.0.1"));
    LocalityTier lt1 = new LocalityTier("node", "localhost");
    LocalityTier lt2 = new LocalityTier("node", "127.0.0.1");

    Configuration.set(PropertyKey.LOCALITY_COMPARE_NODE_IP, "true");
    assertTrue(lt1.matches(lt2));

    Configuration.set(PropertyKey.LOCALITY_COMPARE_NODE_IP, "false");
    assertFalse(lt1.matches(lt2));
  }
```

Reply:
That’s an excellent of testing that case, thanks!

Reply:
Done